lib/opal/nodes/literal.rb

```diff
@@ -216,10 +216,14 @@ class XStringNode < Base
       handle :xstr
 
       def compile
-        children.each do |child|
+        last_index = children.size - 1
+        children.each_with_index do |child, index|
           case child.type
           when :str
             value = child.loc.expression.source
+            if last_index == index
+              value = detect_and_remove_trailing_semicolon(value, child)
+            end
```

Comment:
instead of checking for `last_index` why not check for `expr?` and `recvr?` ([example](https://github.com/opal/opal/blob/66849b5da5c54236eea5a0e5eaf95ba06a744fe3/lib/opal/nodes/if.rb#L46))

that way we can ignore the problem if the line is not used for something else